### PR TITLE
Numeric Field

### PR DESCRIPTION
**Pull request by agriffard**
Numeric Field implementation.

Allow to add a field a specify various settings:
Required
Hint
Min, Max, default value
Scale (= how many decimals)

Will sill need some work when localization and culture site setting will be available.

### CODE REVIEW

File: src/OrchardCore.Modules/Orchard.ContentFields/Drivers/NumericFieldDriver.cs [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+            .Location("Content")
+            .Location("SummaryAdmin", "");
+        }
+
+        public override IDisplayResult Edit(NumericField field, BuildFieldEditorContext context)
+        {
+            return Shape<EditNumericFieldViewModel>("NumericField_Edit", model =>
+            {
+                var settings = context.PartFieldDefinition.Settings.ToObject<NumericFieldSettings>();
+                model.Value = (field.IsNew() && field.Value == null) ? settings.DefaultValue : Convert.ToString(field.Value, _cultureInfo);
```
Comment (sebastienros) [inline src/OrchardCore.Modules/Orchard.ContentFields/Drivers/NumericFieldDriver.cs]: I am thinking we should maybe provide the New/Update boolean in the EditorContext, we will often have this case, either in fields and parts. Same thing in the Update method.

File: src/OrchardCore.Modules/Orchard.ContentFields/Drivers/NumericFieldDriver.cs [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+
+                var settings = context.PartFieldDefinition.Settings.ToObject<NumericFieldSettings>();
+
+                field.Value = null;
+
+                if (string.IsNullOrWhiteSpace(viewModel.Value))
+                {
+                    if (settings.Required)
+                    {
+                        updater.ModelState.AddModelError(Prefix, T["The {0} field is required.", T[context.PartFieldDefinition.DisplayName()]]);
```
Comment (sebastienros) [inline src/OrchardCore.Modules/Orchard.ContentFields/Drivers/NumericFieldDriver.cs]: Don't localize user strings.

**Comment by sebastienros** [comment]
Do you want to add support for custom editors (previously known as flavors)? Like the text field? It would allow others to provide sliders or, +/- icons, ... YOU don't have to implement these, just support and keep the default implementation as the default editor.

**Comment by agriffard** [comment]
OK, I will add support to other editors and options.

**Comment by agriffard** [comment]
I added an Editor setting and a Default and a Range option.

But now, the settings are not saved anymore. It is a behavior I had already seen. See #551.

The line 22 in NumericFieldSettingsDriver.cs doesn't updates the model:
`await context.Updater.TryUpdateModelAsync(model, Prefix);`

**Comment by agriffard** [comment]
Added Bootstrap slider and Jquery UI spinner.
Spinner doesn't show the up/down icons and is not well displayed in the bootstrap form.

Still need to figure out why the settings are not well bound if the Editor in is a `<select>`. See #654.

**Comment by agriffard** [comment]
To fix:
- If a textbox and a slider are displayed, a change in the textbox should also update the slider position.
- With a slider or a range input, the 'live preview' works only if the value in the textbox is changed and Tab pressed but not if the slider is moved.

**Comment by agriffard** [comment]
If I use an input `<input type="number" asp-for="Value" />` the decimal value is loaded with '0,00' but the number expect a format like this '0.00'.

**Comment by sebastienros** [comment]
@agriffard Feel free to merge when you are ready

**Pull request merged**